### PR TITLE
[conductor] Tag engine versions

### LINKED ISSUE #103423: [conductor] Tag flutter engine releases
Add logic to the conductor tool to tag flutter/engine releases.

This will enable us to utilize the github api to generate release notes and leave room for other automations. 

CC/ @christopherfujino @CaseyHillers

### PR DESCRIPTION
**Pull request by CaseyHillers**
Fixes https://github.com/flutter/flutter/issues/103423

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: dev/conductor/core/lib/src/next.dart [hunk 1]
```diff
@@ -264,21 +264,33 @@ class NextContext extends Context {
...
+            upstreamRemote: frameworkUpstream,
             previousCheckoutLocation: state.framework.checkoutPath,
         );
-        final String headRevision = await framework.reverseParse('HEAD');
+        final String frameworkHead = await framework.reverseParse('HEAD');
+        final Remote engineUpstream = Remote(
+            name: RemoteName.upstream,
+            url: state.engine.upstream.url,
+        );
+        final FrameworkRepository engine = FrameworkRepository(
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/next.dart:283]:
```suggestion
        final FrameworkRepository engine = EngineRepository(
```

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:283]: and then you'll have to move the `tag()` method from `FrameworkRepository` to `Repository` (this should be safe)

Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart]: Thanks, done!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**